default/python/AI/ProductionAI.py

```diff
@@ -1690,12 +1690,14 @@ def build_ship_facilities(bld_name, top_locs=None):
         return
     queued_bld_locs = [element.locationID for element in empire.productionQueue if element.name == bld_name]
     if bld_name in AIDependencies.SYSTEM_SHIP_FACILITIES:
-        current_locs = get_systems_with_building(bld_name)
+        current_locs = get_systems_with_facilities(bld_name)
         current_coverage = current_locs.union(universe.getPlanet(planet_id).systemID for planet_id in queued_bld_locs)
         open_systems = set(
             universe.getPlanet(pid).systemID for pid in get_best_pilot_facilities("BLD_SHIPYARD_BASE")
         ).difference(current_coverage)
-        try_systems = open_systems.intersection(get_systems_with_building(prereq_bldg)) if prereq_bldg else open_systems
+        try_systems = (
+            open_systems.intersection(get_systems_with_facilities(prereq_bldg)) if prereq_bldg else open_systems
```

Comment:
Maybe move `if` to the upper level and simplify case `open_systems.intersection(open_systems)`.

Reply:
Well, I did not touch that line, except that I renamed the function and black did some reformatting.
Real changes in this file will come later.
Btw., would you be available for a chat this evening, e.g. 4pm GMT (7pm Estonian time)? Or maybe even now, if you read it quickly.

Reply:
I am not a quick reader. 

@Grummel7 , I'll be in our weekly call meeting for a while.  If you have time jump in.